packages/protocol/contracts/tokenvault/ERC1155Vault.sol

```diff
@@ -170,36 +149,20 @@ contract ERC1155Vault is BaseNFTVault, ERC1155ReceiverUpgradeable {
     {
         checkRecallMessageContext();
 
-        (CanonicalNFT memory nft,,, uint256[] memory tokenIds, uint256[] memory amounts) =
+        (CanonicalNFT memory ctoken,,, uint256[] memory tokenIds, uint256[] memory amounts) =
             abi.decode(message.data[4:], (CanonicalNFT, address, address, uint256[], uint256[]));
 
-        if (nft.addr == address(0)) revert VAULT_INVALID_TOKEN();
+        if (ctoken.addr == address(0)) revert VAULT_INVALID_TOKEN();
 
-        unchecked {
-            if (bridgedToCanonical[nft.addr].addr != address(0)) {
-                for (uint256 i; i < tokenIds.length; ++i) {
-                    BridgedERC1155(nft.addr).mint(message.owner, tokenIds[i], amounts[i]);
-                }
-            } else {
-                for (uint256 i; i < tokenIds.length; ++i) {
-                    ERC1155(nft.addr).safeTransferFrom({
-                        from: address(this),
-                        to: message.owner,
-                        id: tokenIds[i],
-                        amount: amounts[i],
-                        data: ""
-                    });
-                }
-            }
-        }
-        // Send back Ether
+        // Transfer the ETH and tokens back to the owner
+        _transferTokens(ctoken, message.owner, tokenIds, amounts);
         message.owner.sendEther(message.value);
 
         // Emit TokenReleased event
         emit TokenReleased({
             msgHash: msgHash,
             from: message.owner,
-            token: nft.addr,
+            token: ctoken.addr,
```

Comment:
shouldn't it be the addr? (returned by `_transferTokens(ctoken, message.owner, tokenIds, amounts);` )

Reply:
Yeah I stuck with the previous behavior where the canonical token is the one that is always logged. But indeed using the actual token address on the chain probably better.